basis/ui/gadgets/editors/editors.factor

```diff
@@ -213,9 +213,37 @@ M: editor draw-gadget*
         [ draw-lines ] [ draw-caret ] bi
     ] with-variable ;
 
+<PRIVATE
+
+: row,col-dim ( font r c -- dim )
+    "" resize-string [ drop CHAR: x ] map
+    swap "" resize-string [ drop CHAR: \n ] map 
+    append text-dim ;
+
+: min-dim ( font editor -- dim )
+    [ min-rows>> 0 [ and ] most ]
+    [ min-cols>> 0 [ and ] most ] bi
+    row,col-dim ;
+    
+: max-dim ( font editor -- dim )
+    ! hopefully no one goes over 5000
+    [ max-rows>> 5000 [ and ] most ]
+    [ max-cols>> 5000 [ and ] most ] bi
+    row,col-dim ;
+    
+: txt-dim ( font editor -- dim )
+    control-value text-dim ;
+    
+PRIVATE>
+
 M: editor pref-dim*
     ! Add some space for the caret.
-    [ font>> ] [ control-value ] bi text-dim { 1 0 } v+ ;
+    ! Now respects min/max-row/col parameters
+    [ font>> ] keep
+    [ max-dim ]
+    [ txt-dim ]
```

Comment:
I realize you are lining up with min and max but we try not to do too much abbreviation.